service_container/factories.rst

```diff
@@ -330,12 +343,19 @@ Factory definition::
 
     // src/Factories/DeliveryFactory.php
     namespace App\Factories;
+	
+    use App\Deliveries\DeliveryInterface;
+    use RuntimeException;
     
     abstract class DeliveryFactory
     {
         public static function create($deliveryMethod)
```

Comment:
```suggestion
        public static function create(DeliveryInterface $deliveryMethod)
```
Afterwards the manuall check + exception can be removed

Reply:
`$deliveryMethod` is not an object, but a _string_, which is passed in services.yaml:
```
pp\Deliveries\DHL:
      factory:   ['@App\Factories\DeliveryFactory', create]
      arguments: ['App\Deliveries\DHL']   #<-- $deliveryMethod is string 'App\Deliveries\DHL'
```
And _object_ is created basing on this string - this is an approach of factory method I've used.

Let me show you very simple code snippet, which I used as a base for my example creation:

```
abstract class Animal
{
    // factory method
    public static function initial($animal)
    {
        return new $animal();
    }
    abstract public function voice();
}

class Lion extends Animal
{
    public function voice()
    {
        echo 'I am a lion <br />' . PHP_EOL;
    }
}

class Cat extends Animal
{
    public function voice()
    {
        echo 'I am a cat <br />' . PHP_EOL;
    }
}

// As you see, string is passed to factory.
$animal1 = Animal::initial('Lion');
$animal2 = Animal::initial('Cat');

$animal1->voice();
$animal2->voice();
```

My example for factory in Symfony is just to show, how it works _with settings in services.yaml_ and without `implements` key word. Every programmer will be able to run the example, but they also will be able to _modify_ it at their discretion.